vm/src/vm/vm_core.rs

```diff
@@ -1024,6 +1026,45 @@ impl VirtualMachine {
             Err(TraceError::TraceNotRelocated)
         }
     }
+
+    /// Returns a list of addresses of memory cells that constitute the public memory.
+    pub fn get_public_memory_addresses(&self) -> Result<Vec<(usize, &usize)>, VirtualMachineError> {
+        if let Some(relocation_table) = &self.relocation_table {
+            Ok(self.segments.get_public_memory_addresses(relocation_table))
+        } else {
+            Err(MemoryError::UnrelocatedMemory.into())
+        }
+    }
+
+    pub fn get_memory_segment_addresses(
+        &self,
+    ) -> Result<HashMap<&'static str, (usize, Option<usize>)>, VirtualMachineError> {
+        let relocation_table = self
+            .relocation_table
+            .as_ref()
+            .ok_or(MemoryError::UnrelocatedMemory)?;
+
+        let relocate = |segment: (usize, Option<usize>)| -> Result<(usize, Option<usize>), VirtualMachineError> {
+            let (index, stop_ptr_offset) = segment;
+            let base = relocation_table
+                .get(index)
+                .ok_or(VirtualMachineError::RelocationNotFound(index))?;
+            Ok((
+                *base,
+                stop_ptr_offset.map(|s| base + s),
```

Comment:
We could return a `Runner::Error(NoStopPointer)` here if stop_ptr_offset is None and avoid handling an Option in the following code. It doesn't make much sense to return an air_public_input with these values missing

Reply:
Done!